File: app/src/main/java/ru/alek/a22bytenewsapp/MainApp.java
package ru.alek.a22bytenewsapp;

import android.app.Application;

import ru.alek.a22bytenewsapp.data.network.RetrofitManager;
import ru.alek.a22bytenewsapp.data.network.interfaces.api.ApiManager;

public class MainApp extends Application {

    private static MainApp instance;
    private final ApiManager api;

    private MainApp(){
        RetrofitManager retrofitManager = new RetrofitManager();
        api = retrofitManager.getApi();
    }

    public static MainApp getInstance() {
        if(instance == null)  {
            synchronized (MainApp.class){
                if(instance == null){
                    instance = new MainApp();
                }
            }
        }
        return instance;
    }

    public ApiManager getApi() {
        return api;
    }
}

File: app/src/main/java/ru/alek/a22bytenewsapp/MainApp.kt
package ru.alek.a22bytenewsapp

import android.app.Application
import ru.alek.a22bytenewsapp.data.network.NetworkHandler

object MainApp: Application() {

    private lateinit var instance: MainApp

    fun getInstance(): MainApp {
        instance = MainApp
        return instance
    }

    val api = NetworkHandler.api

}